agent/consul/state/config_entry.go

```diff
@@ -102,6 +103,10 @@ func (s *Snapshot) ConfigEntries() ([]structs.ConfigEntry, error) {
 
 // ConfigEntry is used when restoring from a snapshot.
 func (s *Restore) ConfigEntry(c structs.ConfigEntry) error {
+	// only set the ModifyID if it's empty, otherwise use the one from the snapshot
+	if c.ModifyID() == "" {
+		c.SetModifyID(ulid.Make().String())
```

Comment:
This is bad, because it's going to be different on each of the servers so every leader election will cause the same issue you are working to avoid.

Reply:
If you're going to do this, you will have to do it deterministically, and manufacture a `[16]byte` that is non-overlapping with any ulid produced in the future. You may be able to cheat and use the raft modify index creatively with some fixed time element to create your own ulids without any hint of clocks or randomness.

Reply:
Do we do a restore on all the servers? I thought restore is only done in one server and then it replicate

Reply:
restore is run separately on all servers. They only exchange the full snapshot if they do not have one on disk already, or it's too old compared to the raft log

Reply:
Why not use something more deterministic than a ULID. You could hash the structure similar to what we already do for ACL tokens. Because the hash is deterministic you can do this in the state store and hit both snapshot restore and new log replication paths.

It sounds like the randomness and time elements of the ULID aren't actually being relied upon and so the hash based approach should meet the need just as well.

Reply:
yes that's right, I thought about the hash approach but to be used on compare time. I think a hash on write would be even better as we can have a deterministic ID but also avoid the extra CPU load when comparing. 
I will take a stab on that and let you know to have a second pass of review